src/main/java/org/isf/medicalstock/manager/MovStockInsertingManager.java

```diff
@@ -352,16 +350,33 @@ protected Movement prepareChargingMovement(Movement movement, boolean checkRefer
 
 	/**
 	 * Retrieves the {@link Lot}.
+	 * 
 	 * @param lotCode - the lot code.
 	 * @return the retrieved {@link Lot}.
 	 * @throws OHServiceException if an error occurs during the check.
 	 */
 	public Lot getLot(String lotCode) throws OHServiceException {
 		return ioOperations.getLot(lotCode);
 	}
-	
+
+	/**
+	 * Update the list of {@link Lot}s.
+	 * 
+	 * @param lots - the list of lots.
+	 * @return the list of {@link Lot}s updated.
+	 * @throws OHServiceException if an error occurs during the check.
+	 */
+	@Transactional
+	public List<Lot> updateLot(List<Lot> lots) throws OHServiceException {
+		for (Lot lot : lots) {
+			lot = ioOperations.updateLot(lot);
+		}
+		return lots;
```

Comment:
This doesn't look like it is returning a list of the updated `lots` but rather the original list of `lots`.   Most likely they are the same but shouldn't the returned `lot` from `ioOperations.updateLot(lot);` be added to a new `List` of `Lot` objects and that be returned?    As it is not the return value from`ioOperations.updateLot(lot)` is not used and thrown away.

Reply:
Nice catch! I was testing without a new list and it worked (the returned lot is indeed keeping the same reference in memory, although it could not be always true), then I left it like that. Fixed, thanks!